### ISSUE TITLE
Ubuntu Launchpad CI build failling

### ISSUE THREAD

**Issue by jfmcarreira** [fhackenberger/ktikz#73]
Fix the CI service of the ubuntu launchpad

**Comment by jfmcarreira** [comment]
@alexfikl Do you use ubuntu for development? I am having problems making the build pass using qmake. however the github action you added passes here.

**Comment by alexfikl** [comment]
No, I'm afraid I'm using arch (and their packaging is quite different from what I recall).

Like you said the CI uses Ubuntu, the 22.04 LTS from what it says on https://github.com/actions/runner-images?tab=readme-ov-file#available-images, so it should work. What errors are you seeing?

Also, what's the "Ubuntu Launchpad CI"? Are the logs from that available?

**Comment by jfmcarreira** [comment]
the error is pretty easy to spot. :)
https://launchpadlibrarian.net/726492683/buildlog_ubuntu-jammy-amd64.ktikz_0.13.1-202404262122-5696d92~ubuntu22.04.1_BUILDING.txt.gz

In case it requires login:
<img width="1716" alt="image" src="https://github.com/fhackenberger/ktikz/assets/5453402/219ac608-aa41-4367-b3f7-fd23d9a18446">

However, I do not get why it is not finding poppler.

I am setting up a ubuntu virtual machine to test the issue. The other day I did a bunch of force push, since I was playing around to see if i could fix it. I also changed some minor things to make it pass.

I am targetting for at least Ubuntu 22.04.

Maybe it could be interesting if we have a package for arch :) not sure how hard it is.

**Comment by alexfikl** [comment]
> the error is pretty easy to spot. :) [launchpadlibrarian.net/726492683/buildlog_ubuntu-jammy-amd64.ktikz_0.13.1-202404262122-5696d92~ubuntu22.04.1_BUILDING.txt.gz](https://launchpadlibrarian.net/726492683/buildlog_ubuntu-jammy-amd64.ktikz_0.13.1-202404262122-5696d92~ubuntu22.04.1_BUILDING.txt.gz)

Yep, that's very clear! The Github CI isn't doing anything special from what I can tell, so not sure why it doesn't find poppler includes on the Launchpad version. Especially since they seem to be using the same ubuntu version..

> I am setting up a ubuntu virtual machine to test the issue. The other day I did a bunch of force push, since I was playing around to see if i could fix it. I also changed some minor things to make it pass.

I can try the docker image locally to see if there's any difference there :\

> Maybe it could be interesting if we have a package for arch :) not sure how hard it is.

There's already an official package for Arch: https://archlinux.org/packages/extra/x86_64/ktikz/ :grin:

**Comment by alexfikl** [comment]
Just from looking at the logs, the Github CI clearly has a bunch of (repeated) poppler includes that the Launchpad run is missing:
```
# Github
g++ -c -pipe -I/usr/include/poppler/qt5 -I/usr/include/poppler -I/usr/include/poppler/qt5 -I/usr/include/poppler -I/usr/include/poppler/qt5 -I/usr/include/poppler -O2 -D_REENTRANT -Wall -Wextra [...]
# Launchpad
g++ -c -pipe -O2 -D_REENTRANT -Wall -Wextra [...]
```

The includes should be found in `qmake/findpoppler.pri` using `pkg-config`:
```
    POPPLERINCLUDES = $$system($$PKG_CONFIG --cflags poppler-qt5)
```

**Comment by jfmcarreira** [comment]
Yeah i noticed that too. seems like in the ubuntu launchpad the pkg-config for the poppler is not giving the right result

**Comment by jfmcarreira** [comment]
I think it is a missing dependency on pkg-config

**Comment by jfmcarreira** [comment]
Solved the build issue. Now just adding back a dependency i removed and didnt allow to process icons :)

**Comment by fhackenberger** [comment]
I'd close this then. Thank you!
